File: scala/digitalwallet/app/src/main/scala/adapters/Logger.scala
package adapters

class Logger {
  def error(message: String, arguments: Option[Map[String, String]] = None): Unit = {
    println(message)
    arguments.foreach { args =>
      println("Arguments:")
      args.foreach { case (key, value) =>
        println(s"$key: $value")
      }
    }
  }
}

File: kotlin/digital-wallet/src/main/kotlin/digitalwallet/adapters/Logger.kt
package digitalwallet.adapters

class Logger {
    fun error(message: String, arguments: Map<String, String>? = null) {
        println(message)
        arguments?.let {
            println("Arguments:")
            it.forEach { (key, value) ->
                println("$key: $value")
            }
        }
    }
}